lib/cdo/google/sheet.rb

```diff
@@ -21,29 +21,52 @@ def initialize(document_key)
       @document_key = document_key
     end
 
-    def export(sheet_name:, rows:)
+    def export(tab_name:, data:)
       # Write exported data to a sheet in the document
-      @drive.update_sheet rows, @document_key, "#{sheet_name} (auto)"
+      @drive.update_sheet data, @document_key, "#{tab_name} (auto)"
 
       # Write new metadata to a second sheet in the document
       last_updated = DateTime.now.in_time_zone ActiveSupport::TimeZone.new "Pacific Time (US & Canada)"
       metadata = <<~META.split("\n").map {|line| [line]}
         AUTOMATION METADATA
 
-        The tabs "#{sheet_name} (auto)" and "#{sheet_name}_meta (auto)" are auto-generated;
+        The tabs "#{tab_name} (auto)" and "#{tab_name}_meta (auto)" are auto-generated;
         Any edits you make to them (besides formatting) may be lost.
 
         Last updated: #{last_updated.strftime '%Y-%m-%d %l:%M%P GMT%:::z'}
         Written by: #{CDO.gdrive_export_secret.client_email}
-        Rows: #{rows.length - 1}
+        Rows: #{data.length - 1}
 
         Parts of this Google Sheet are auto-populated from our live application by an automated process.
         The sheet is shared with a \"service account\" that updates it on the application's behalf.
         (Technical Details: https://github.com/code-dot-org/code-dot-org/pull/32597)
       META
-      @drive.update_sheet metadata, @document_key, "#{sheet_name}_meta (auto)"
+      @drive.update_sheet metadata, @document_key, "#{tab_name}_meta (auto)"
     end
 
+    def notify_of_external_sharing(dcdo_key=nil)
```

Comment:
I'd suggest we have this just accept an array of allowed emails, and leave it up to the calling methods to decide how they get that information. It might mean that for all our current methods they have to repeat the minor pattern of retrieving an array from a DCDO, but I think that's better than requiring that all future callers implement the same pattern.

Reply:
Good idea! Updating.

Reply:
Re: document key, I think you're right that there's enough there to identify what's wrong from the HB trace. Removing.